tfplan/json_plan.go

```diff
@@ -59,111 +50,27 @@ type jsonResourceChange struct {
 }
 
 // isCreate returns true if the action on the resource is ["create"].
-func (c *jsonResourceChange) isCreate() bool {
+func (c *ResourceChange) isCreate() bool {
 	return len(c.Change.Actions) == 1 && c.Change.Actions[0] == "create"
 }
 
-type childModule struct {
-	Address      string         `json:"address"`
-	Resources    []jsonResource `json:"resources"`
-	ChildModules []childModule  `json:"child_modules"`
+// compatibility shim until ResourceChange is expected by all callers.
+func (c *ResourceChange) Kind() string {
+	return c.Type
 }
 
-// jsonResource represent single Terraform resource definition.
-type jsonResource struct {
-	Module       string                 `json:"module"`
-	Name         string                 `json:"name"`
-	Address      string                 `json:"address"`
-	Mode         string                 `json:"mode"`
-	Kind         string                 `json:"type"`
-	ProviderName string                 `json:"provider_name"`
-	Values       map[string]interface{} `json:"values"`
+// ComposeTF12Resources is a thin wrapper around ReadResourceChanges as a compatibility shim.
+func ComposeTF12Resources(data []byte, schemas map[string]*schema.Resource) ([]ResourceChange, error) {
```

Comment:
This is something we can eliminate and we just don't want to do that in this PR right?

Reply:
Deleting this would cause an incompatibility as described [here](https://docs.google.com/document/d/1_FDdCuqs3hpaceZRXnKfJJ0VgdsJzuKYsmOEA0AZIi8/edit#heading=h.gijrsdjw51q5). TLDR we would need to update calling code prior to removing this; removing this function would be a breaking change. (But yes, we can eliminate it after the calling code is updated.)